flax/linen/recurrent.py

```diff
@@ -500,6 +500,112 @@ def num_feature_axes(self) -> int:
+    gate_fn: activation function used for gates (default: sigmoid).
+    activation_fn: activation function used for output and memory update
+      (default: tanh).
+    kernel_init: initializer function for the kernels that transform
+      the input (default: lecun_normal).
+    recurrent_kernel_init: initializer function for the kernels that transform
+      the hidden state (default: initializers.orthogonal()).
+    forget_bias_init: initializer for the bias parameters of the forget gate.
+      The default is set to initializers.ones_init() because this prevents
+      vanishing gradients. See https://proceedings.mlr.press/v37/jozefowicz15.pdf,
+      section 2.2 for more details.
+    activation_bias_init: initializer for the bias parameters of the activation
+      output (default: initializers.zeros_init()).
+    dtype: the dtype of the computation (default: None).
+    param_dtype: the dtype passed to parameter initializers (default: float32).
+  """
+  features: int
+  gate_fn: Callable[..., Any] = sigmoid
+  activation_fn: Callable[..., Any] = tanh
+  kernel_init: initializers.Initializer = default_kernel_init
+  recurrent_kernel_init: initializers.Initializer = initializers.orthogonal()
+  forget_bias_init: initializers.Initializer = initializers.ones_init()
+  activation_bias_init: initializers.Initializer = initializers.zeros_init()
+  dtype: Optional[Dtype] = None
+  param_dtype: Dtype = jnp.float32
+  carry_init: initializers.Initializer = initializers.zeros_init()
+
+  @compact
+  def __call__(self, carry, inputs):
+    """Minimal gated unit (MGU) cell.
+
+    Args:
+      carry: the hidden state of the MGU cell,
+        initialized using `MGUCell.initialize_carry`.
+      inputs: an ndarray with the input for the current time step.
+        All dimensions except the final are considered batch dimensions.
+
+    Returns:
+      A tuple with the new carry and the output.
+    """
+    h = carry
+    hidden_features = h.shape[-1]
+    # input and recurrent layers are summed so only one needs a bias.
+    dense_h = partial(
+        Dense,
+        features=hidden_features,
+        use_bias=False,
+        dtype=self.dtype,
+        param_dtype=self.param_dtype,
+        kernel_init=self.recurrent_kernel_init,
+        bias_init=self.activation_bias_init,
+    )
+    dense_i = partial(
+        Dense,
+        features=hidden_features,
+        use_bias=True,
+        dtype=self.dtype,
+        param_dtype=self.param_dtype,
+        kernel_init=self.kernel_init,
+    )
+    f = self.gate_fn(dense_i(name='if', bias_init=self.forget_bias_init)(inputs) + dense_h(name='hf')(h))
+    # add bias because the linear transformations aren't directly summed.
+    n = self.activation_fn(
+        dense_i(name='in', bias_init=self.activation_bias_init)(inputs) + f * dense_h(name='hn', use_bias=True)(h)
+    )
+    new_h = (1.0 - f) * n + f * h
+    return new_h, new_h
+
+  @nowrap
+  @deprecation_method_decorator
```

Comment:
Lets merge #3425 before this PR so we can remove this decorator.

Reply:
done